src/point_ops.rs

```diff
@@ -51,7 +51,47 @@ impl PointOp {
 
     /// Verifies a batch of point operations using one multi-scalar multiplication
     pub fn verify_batch(batch: &[PointOp]) -> Result<(), VMError> {
-        unimplemented!()
+        let gens = PedersenGens::default();
+
+        let length: usize = batch.iter().map(|p| p.arbitrary.len()).sum();
+
+        let mut weights: Vec<Scalar> = Vec::with_capacity(length + 2);
+        let mut points: Vec<Option<RistrettoPoint>> = Vec::with_capacity(length + 2);
+
+        // Add base points
+        points.push(Some(gens.B));
+        points.push(Some(gens.B_blinding));
+        weights.push(Scalar::default());
+        weights.push(Scalar::default());
+
+        // Iterate over every point, adding both weights and points to
+        // our arrays
+        for p in batch.iter() {
+            // Sample free variable e
+            let e = Scalar::random(&mut rand::thread_rng());
```

Comment:
this creates a new RNG instance in each iteration. I'd move it out of the loop and reuse the RNG:

```
let mut rng = rand::thread_rng();
...
let e = Scalar::random(&mut rng);
```